src/lib/registration/registration.rb

```diff
@@ -167,24 +167,19 @@ def migration_products(installed_products)
       migrations
     end
 
-    # Get the list of updates for a given product
+    # Get the list of updates for a base product or self_update_id if defined
     #
-    # @param [Hash] Hash containing the product description.
-    #               Description should contain "name", "arch",
-    #               "version" and "release_type".
     # @return [Array<String>] List of URLs of updates repositories.
     #
     # @see SwMgmt.base_product_to_register
     # @see SwMgmt.remote_product
     # @see SUSE::Connect::Yast.list_installer_updates
-    def get_updates_list(product = nil)
-      if product.nil?
-        product = SwMgmt.base_product_to_register
-        id = Yast::ProductFeatures.GetStringFeature("globals", "self_update_id")
-        if !id.empty?
-          log.info "Using self update id from control file #{id.inspect}"
-          product["name"] = id
-        end
+    def get_updates_list
+      product = SwMgmt.base_product_to_register
+      id = Yast::ProductFeatures.GetStringFeature("globals", "self_update_id")
+      if !id.empty?
+        log.info "Using self update id from control file #{id.inspect}"
+        product["name"] = id
```

Comment:
I'd add a comment saying that we expect all products on the medium to be in the same version so we can replace only the name attribute. This is an expectation which is not obvious from the code.

Reply:
Yes, makes sense for me.